### PR TITLE
docs: python openapi client

### LINKED ISSUE #1701: OpenAPI3 python client and documentation
We used Boto for our Python client for openapi.
As we move to use openapi v3, Boto will no longer work and we will need to document and use the python generated code.

### PR DESCRIPTION
**Pull request by nopcoder**
Closes #1701

Document how to create python code and use lakeFS API (replace bravado after moving to use openapi v3)

### CODE REVIEW

**Comment by nopcoder** [comment]
Switched to draft until #1711 is complete which will require docs update.

File: docs/using/python.md [hunk 1]
```diff
@@ -10,164 +10,161 @@ has_children: false
 # Calling the lakeFS API from Python
 {: .no_toc }
 
-The [lakeFS API](../reference/api.md){: target="_blank" } is OpenAPI 2.0 compliant, allowing the dynamic generation of clients from multiple languages.
+The [lakeFS API](../reference/api.md){: target="_blank" } is OpenAPI 3.0 compliant, allowing the generation of clients from multiple languages.
 
-For Python, this example uses [Bravado](https://github.com/Yelp/bravado){: target="_blank" }
-which generates a dynamic client at runtime, from an OpenAPI definition served by a lakeFS server.
+For Python, this example uses [OpenAPI Generator](hhttps://openapi-generator.tech){: target="_blank" }
```
Comment (itaiad200) [inline docs/using/python.md:15]:
```suggestion
For Python, this example uses [OpenAPI Generator](https://openapi-generator.tech){: target="_blank" }
```

File: docs/using/python.md [hunk 2]
```diff
@@ -10,164 +10,161 @@ has_children: false
...
+which can generates Python client code for the OpenAPI definition served by a lakeFS server.
```
Comment (itaiad200) [inline docs/using/python.md:16]:
```suggestion
that generates Python client code for the OpenAPI definition served by a lakeFS server.
```

File: docs/using/python.md [hunk 3]
```diff
@@ -10,164 +10,161 @@ has_children: false
...
 ```shell
-pip install bravado==10.6.2
+pip install 'lakefs==<lakeFS version>'
 ```
 
-At the time of writing this guide, the current stable bravado release is `10.6.2`.
-
-
-## Locating the OpenAPI definition for our installation
+The package is available from version 0.34.0.
```
Comment (itaiad200) [inline docs/using/python.md:33]: Where did you come up with this version?

File: docs/using/python.md [hunk 4]
```diff
@@ -10,164 +10,161 @@ has_children: false
...
-# repository(creation_date=1599560048, default_branch='main', id='example-repo', storage_namespace='s3://storage-bucket/repos/example-repo')
+# {'creation_date': 1617532175,
+#  'default_branch': 'main',
+#  'id': 'example-repo',
+#  'storage_namespace': 's3://storage-bucket/repos/example-repo'}
 ```
 
 ### Creating a branch, uploading files, committing changes
 
 List current branches:
```
Comment (itaiad200) [inline docs/using/python.md:71]: Why `current`? Do you hold the state somewhere?

**Review by itaiad200 — APPROVED** [review]
Looks great! see minor suggestions.

File: docs/using/python.md (hunk 3 above)
Comment (nopcoder) [reply docs/using/python.md:33]: assumed it will be the next release - but I'll have to update it after the release

File: docs/using/python.md [hunk 5]
```diff
@@ -10,164 +10,161 @@ has_children: false
 # Calling the lakeFS API from Python
 {: .no_toc }
 
-The [lakeFS API](../reference/api.md){: target="_blank" } is OpenAPI 2.0 compliant, allowing the dynamic generation of clients from multiple languages.
+The [lakeFS API](../reference/api.md){: target="_blank" } is OpenAPI 3.0 compliant, allowing the generation of clients from multiple languages.
 
-For Python, this example uses [Bravado](https://github.com/Yelp/bravado){: target="_blank" }
-which generates a dynamic client at runtime, from an OpenAPI definition served by a lakeFS server.
+For Python, this example uses [lakeFS's python package](https://pypi.org/project/lakefs-client/){: target="_blank" }.
+The lakefs-client pacakge was created by [OpenAPI Generator](https://openapi-generator.tech){: target="_blank" } using our OpenAPI definition served by a lakeFS server.
```
Comment (itaiad200) [inline docs/using/python.md:16]: I would add a note that this is the recommended client package for python, but other solutions are valid too (Bravado, simple HTTP client)

File: docs/using/python.md [hunk 6]
```diff
@@ -10,164 +10,161 @@ has_children: false
...
 ```shell
-pip install bravado==10.6.2
+pip install 'lakefs_client==<lakeFS version>'
 ```
 
-At the time of writing this guide, the current stable bravado release is `10.6.2`.
-
-
-## Locating the OpenAPI definition for our installation
+The package is available from version > 0.33.1.
```
Comment (itaiad200) [inline docs/using/python.md:33]: Better to be concrete here. 
```suggestion
The package is available from version >= 0.34.0.
```
?

File: docs/using/python.md [hunk 7]
```diff
@@ -10,164 +10,161 @@ has_children: false
...
+from lakefs_client import models
+from lakefs_client.client import LakeFSClient
 
-http_client = RequestsClient()
-http_client.set_basic_auth('localhost:8000', 'AKIAIOSFODNN7EXAMPLE', 'wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY')
-client = SwaggerClient.from_url('http://localhost:8000/swagger.json', http_client=http_client)
+configuration = lakefs_client.Configuration()
+configuration.username = 'AKIAIOSFODNN7EXAMPLE'
+configuration.password = 'wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY'
+configuration.host = 'http://localhost:8000/api/v1'
```
Comment (itaiad200) [inline docs/using/python.md:48]: Add a comment describing that these are the values for your lakeFS installation.
Might be trivial to us, but users often mix up these values.

**Review by itaiad200 — APPROVED** [review]

File: docs/using/python.md (hunk 5 above)
Comment (nopcoder) [reply docs/using/python.md:16]: I'll add a note that lakeFS can be accessed using any OpenAPI v3 library or directly by HTTP client.
Bravado can't be used - supports only OpenAPI v1 and v2.

**Review by itaiad200 — APPROVED** [review]

**Pull request merged**